lib/models/addon.js

```diff
@@ -448,38 +448,68 @@ Addon.prototype.compileStyles = function(tree) {
-  @param {Tree} tree Templates file tree
+  @param {Tree} addonTree Addon file tree
   @return {Tree} Compiled templates tree
 */
-Addon.prototype.compileTemplates = function(tree) {
+Addon.prototype.compileTemplates = function(addonTree) {
   this._requireBuildPackages();
 
-  if (tree) {
+  var isTemplateInAddon = false;
+
+  var templatePatterns = this.registry.extensionsForType('template').map(function (extension) {
+    return new RegExp('template.' + extension + '$');
+  });
+
+  var standardTemplatesTree = this._treeFor('addon-templates');
+  if (standardTemplatesTree) {
+    isTemplateInAddon = true;
+  }
+  else {
+    var addonTreeFiles = walkSync(addonTree);
+    for (var fileIndex = 0; fileIndex < addonTreeFiles.length; fileIndex++) {
+      for (var patternIndex = 0; patternIndex < templatePatterns.length; patternIndex++) {
+        if (templatePatterns[patternIndex].test(addonTreeFiles[fileIndex])) {
+          isTemplateInAddon = true;
+          break;
+        }
+      }
+
+      if (isTemplateInAddon) {
+        break;
+      }
+    }
+  }
+
+  if (isTemplateInAddon) {
     var plugins = this.registry.load('template');
     if (plugins.length === 0) {
-      throw new SilentError('An `addon/templates` tree was detected, but there ' +
-                            'are no template compilers registered for `' + this.name + '`. ' +
-                            'Please make sure your template precompiler (commonly `ember-cli-htmlbars`) ' +
-                            'is listed in `dependencies` (NOT `devDependencies`) in ' +
-                            '`' + this.name + '`\'s `package.json`.');
+      throw new SilentError('Templates were detected in the addon tree, but there ' +
+      'are no template compilers registered for `' + this.name + '`. ' +
+      'Please make sure your template precompiler (commonly `ember-cli-htmlbars`) ' +
+      'is listed in `dependencies` (NOT `devDependencies`) in ' +
+      '`' + this.name + '`\'s `package.json`.');
     }
 
-    var standardTemplates = new Funnel(tree, {
-      srcDir: '/',
-      destDir: 'modules/' + this.name + '/templates'
-    });
-
-    var includePatterns = this.registry.extensionsForType('template').map(function(extension) {
-      return new RegExp('template.' + extension + '$');
-    });
-
-    var podTemplates = new Funnel(tree, {
-      include: includePatterns,
+    var podTemplatesTree = new Funnel(addonTree, {
+      include: templatePatterns,
+      exclude: [new RegExp('^templates/')],
       destDir: 'modules/' + this.name + '/',
       description: 'Funnel: Addon Pod Templates'
     });
+    var addonTemplatesTree = podTemplatesTree;
+
+    if (standardTemplatesTree) {
+      addonTemplatesTree = mergeTrees([
+        new Funnel(standardTemplatesTree, {
+          srcDir: '/',
+          destDir: 'modules/' + this.name + '/templates',
+          description: 'Funnel: Addon Standard Templates'
+        }),
+        podTemplatesTree
```

Comment:
Pick up the standard addon/templates directory if it exists and merge with the pods based templates.